ledger/ledger-utxo/src/main/kotlin/net/corda/v5/ledger/utxo/ContractStates.kt

```diff
@@ -0,0 +1,86 @@
+package net.corda.v5.ledger.utxo
+
+import net.corda.v5.base.annotations.CordaSerializable
+import java.security.PublicKey
+import java.util.*
+import kotlin.reflect.KClass
+
+/**
+ * Indicates the [Contract] that the current state belongs to.
+ */
+@Retention(AnnotationRetention.RUNTIME)
+@Target(AnnotationTarget.CLASS)
+annotation class BelongsToContract(val value: KClass<out Contract>)
+
+/**
+ * Indicates the URI that references the legal prose associated with the current [Contract].
+ */
+@Target(AnnotationTarget.CLASS)
+@MustBeDocumented
+annotation class LegalProseReference(val value: String)
+
+/**
+ * Defines a contract state.
+ *
+ * A contract state (or just "state") contains opaque data used by a contract program. It can be thought of as a disk
+ * file that the program can use to persist data across transactions.
+ *
+ * States are immutable. Once created they are never updated, instead, any changes must generate a new successor state.
+ * States can be updated (consumed) only once. The notary is responsible for ensuring there is no "double spending" by
+ * only signing a transaction if the input states are all free.
+ *
+ * @property participants The public keys of any participants associated with the current contract state.
+ */
+@CordaSerializable
+interface ContractState {
+    val participants: Set<PublicKey>
+}
+
+/**
+ * Defines a contract state that is identifiable.
+ *
+ * The expectation is that identifiable state implementations will evolve by superseding previous versions of
+ * themselves whilst sharing a common identifier (id).
+ *
+ * @property id The identifier of the current state.
+ */
+interface IdentifiableState : ContractState {
```

Comment:
I would like to leave all these linear state, fungible state, etc models out till we have basic UTXO stuff working on generic extensions of ContractState

Reply:
@mnesbit Do you want these removing, or just no further dev on different `ContractState` types until UTXO fundamentally works?